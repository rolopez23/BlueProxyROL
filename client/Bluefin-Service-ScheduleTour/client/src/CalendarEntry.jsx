/* eslint-disable react/prop-types */

import React from 'react';
import {
  RegularWrapper, SelectedWrapper, Day, Date, Month,
} from './styles/CalendarEntry-style';

const CalendarEntry = ({
  date, clickFn, selected, refs,
}) => {
  const { dayOfWeek, month, dayOfMonth } = date;
  return (
    selected
      ? (
        <SelectedWrapper onClick={() => { clickFn(date); }} ref={refs}>
          <Day className="day">{dayOfWeek}</Day>
          <Date className="date">{dayOfMonth}</Date>
          <Month className="month">{month}</Month>
        </SelectedWrapper>
      )
      : (
        <RegularWrapper onClick={() => { clickFn(date); }} ref={refs}>
          <Day className="day">{dayOfWeek}</Day>
          <Date className="date">{dayOfMonth}</Date>
          <Month className="month">{month}</Month>
        </RegularWrapper>
      )
  );
};

export default CalendarEntry;
