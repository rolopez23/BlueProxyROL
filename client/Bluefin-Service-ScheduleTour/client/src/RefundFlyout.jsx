import React from 'react';
import {
  Wrapper, Box, DownArrow, P, A,
} from './styles/RefundFlyout-style';

const RefundFlyout = () => (
  <Wrapper>
    <Box>
      <P>Buy with a Bluefin Agent and we&apos;ll give you a</P>
      <P>percentage of our commision from the seller.</P>
      <A>Learn More</A>
    </Box>
    <DownArrow />
  </Wrapper>
);

export default RefundFlyout;
