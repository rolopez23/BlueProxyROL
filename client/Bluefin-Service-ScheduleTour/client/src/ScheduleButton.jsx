/* eslint-disable react/prop-types */
import React from 'react';
import Button from './styles/ScheduleButton-style';

// there will be some logic in this component to check whether tours are available that day.
// if tour not available, render "this home is unavailable to tour on this day" rather than button
const ScheduleButton = ({ clickFn }) => (
  <>
    <Button onClick={clickFn}>Schedule Tour</Button>
  </>
);

export default ScheduleButton;
