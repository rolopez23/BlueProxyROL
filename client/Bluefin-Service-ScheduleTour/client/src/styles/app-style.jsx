import styled, { createGlobalStyle } from 'styled-components';

const AppWrapper = styled.div`
  height: 436px;
  width: 313px;
  border: 1px solid black;
  margin-left: 100px;
`;

const Header = styled.h3`
  text-align: center;
  font-size: 1.375rem;
  font-weight: 600;
`;

const Paragraph = styled.div`
  color: #999;
  margin-top: 5px;
  margin-left: 10px;
  text-align: center;
  font-size: 0.75rem;
`;

const CalendarWrapper = styled.div`
  height: 114px;
  width: 313px;
  position: relative
  display: inline-block;
`;

const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Libre+Franklin&display=swap');
  body {
    font-family: 'Libre-Franklin', sans-serif;
  }
`;

const NextButton = styled.div`
  height: 32px;
  width: 32px;
  border: 1px solid #777;
  border-radius: 100%;
  display: inline-block;
  position: relative;
  z-index: 100;
  bottom: 67px;
  left: 279px;
`;

export {
  AppWrapper,
  Header,
  Paragraph,
  CalendarWrapper,
  GlobalStyles,
  NextButton,
};
