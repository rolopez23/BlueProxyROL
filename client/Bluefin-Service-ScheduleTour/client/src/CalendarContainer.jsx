/* eslint-disable react/prop-types */
import React from 'react';
import CalendarEntry from './CalendarEntry';
import {
  Wrapper, EntryWrapper, NextButton, PrevButton,
} from './styles/CalendarContainer-style';


export default class CalendarContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentStart: 0,
      prevButton: false,
      nextButton: true,
      refs: null,
    };
    this.prevHandleClick = this.prevHandleClick.bind(this);
    this.nextHandleClick = this.nextHandleClick.bind(this);
  }
  // these methods allow for button scrolling;


  componentDidMount() {
    const { dates } = this.props;
    const refs = dates.reduce((acc, value) => {
      acc[value.id] = React.createRef();
      return acc;
    }, {});
    this.setState({
      refs,
    });
  }

  prevHandleClick() {
    const { currentStart, nextButton, refs } = this.state;
    this.setState({ currentStart: currentStart - 3 }, () => {
      // have to reset the currentstart variable
      // eslint-disable-next-line
      const { currentStart } = this.state;
      if (!nextButton) {
        this.setState({ nextButton: true });
      }
      refs[currentStart].current.scrollIntoView({
        behavior: 'smooth',
        inline: 'start',
      });
      if (!currentStart) {
        this.setState({ prevButton: false });
      }
    });
  }

  nextHandleClick() {
    const { currentStart, prevButton, refs } = this.state;
    this.setState({ currentStart: currentStart + 3 }, () => {
      // have to reset the currentstart variable
      // eslint-disable-next-line
      const { currentStart } = this.state;
      if (!prevButton) {
        this.setState({ prevButton: true });
      }
      if (currentStart === 9) {
        this.setState({ nextButton: false });
      }
      refs[currentStart].current.scrollIntoView({
        behavior: 'smooth',
        inline: 'start',
      });
    });
  }

  render() {
    const { dates, clickFn, selectedDate } = this.props;
    const { prevButton, nextButton, refs } = this.state;
    return (
      <Wrapper>
        <EntryWrapper>
          {dates.map((date) => {
            // logic:there cant be the same day number in two weeks.
            // So comparing only the day would suffice
            let selected = false;
            if (selectedDate && selectedDate.dayOfMonth === date.dayOfMonth) {
              selected = true;
            }
            return (
              <>
                <span>
                  <CalendarEntry
                    refs={refs ? refs[date.id] : null}
                    key={date}
                    date={{
                      dayOfWeek: date.dayOfWeek,
                      month: date.month,
                      dayOfMonth: date.dayOfMonth,
                    }}
                    clickFn={clickFn}
                    selected={selected}
                  />
                </span>
              </>
            );
          })}
        </EntryWrapper>
        { prevButton
          ? (
            <PrevButton onClick={() => this.prevHandleClick()}>
              <svg viewBox="-13 -20 32 42" fill="#777" transform="rotate(180)">
                <path d="M1.214 11.921L.079 10.786a.268.268 0 0 1 0-.379L4.486 6 .079 1.593a.268.268 0 0 1 0-.38L1.214.08a.268.268 0 0 1 .379 0L7.325 5.81a.268.268 0 0 1 0 .38L1.593 11.92a.268.268 0 0 1-.38 0" fillRule="evenodd" />
              </svg>
            </PrevButton>
          )
          : <></>}
        { nextButton
          ? (
            <NextButton
              left={prevButton ? '245px' : '279px'}
              onClick={() => this.nextHandleClick()}
            >
              <svg viewBox="-12 -10 32 42" fill="#777">
                <path d="M1.214 11.921L.079 10.786a.268.268 0 0 1 0-.379L4.486 6 .079 1.593a.268.268 0 0 1 0-.38L1.214.08a.268.268 0 0 1 .379 0L7.325 5.81a.268.268 0 0 1 0 .38L1.593 11.92a.268.268 0 0 1-.38 0" fillRule="evenodd" />
              </svg>
            </NextButton>
          )
          : <></>}
      </Wrapper>
    );
  }
}
