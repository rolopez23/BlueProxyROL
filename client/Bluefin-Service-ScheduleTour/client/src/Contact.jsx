/* eslint-disable react/prop-types */
import React from 'react';
import { Wrapper, Box1, Box2 } from './styles/Contact-Style';

const Contact = ({ phone, clickFn }) => (
  <Wrapper>
    <Box1 href="#" onClick={clickFn}>Ask A Question</Box1>
    <Box2 href="tel:+I-MISS-KOBE">{ phone }</Box2>
  </Wrapper>
);

export default Contact;
