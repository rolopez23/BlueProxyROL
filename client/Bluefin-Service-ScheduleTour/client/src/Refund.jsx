/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { Wrapper, Text, SVG } from './styles/Refund-style';
import RefundFlyout from './RefundFlyout';

const Refund = ({ price }) => {
  const [toggle, setToggle] = useState(false);

  return (
    <Wrapper>
      { toggle
        ? <RefundFlyout />
        : <></> }
      <Text onClick={() => setToggle(!toggle)}>
        Bluefin Refund: $
        { Math.floor(price * 0.0037) }
      </Text>
      <SVG width="13pt" height="13pt" viewBox="0 0 1000 1000" onClick={() => setToggle(!toggle)}>
        <g transform="translate(0, 1000) scale(0.13000,-0.13000)" stroke="none">
          <path d="M3225 6770 c-106 -10 -297 -37 -360 -51 -369 -83 -557 -149 -865
          -303 -592 -297 -1126 -833 -1421 -1425 -101 -204 -148 -320 -202 -491 -83
          -266 -120 -454 -147 -745 -23 -253 0 -667 51 -890 46 -201 92 -370 115 -422 8
          -17 14 -37 14 -45 0 -7 6 -25 14 -40 8 -14 20 -43 26 -65 25 -73 157 -348 218
          -450 95 -160 230 -347 348 -484 151 -174 396 -399 564 -519 200 -142 323 -212
          605 -345 120 -57 356 -136 530 -179 172 -41 197 -46 315 -62 521 -70 968 -32
          1470 123 152 47 290 104 480 198 368 181 711 449 999 780 120 139 135 158 256
          335 147 214 287 503 376 774 114 348 168 680 168 1036 0 346 -47 650 -156
          1000 -34 108 -85 247 -118 315 -134 283 -195 391 -348 608 -162 231 -510 579
          -738 738 -74 51 -231 153 -283 183 -138 80 -501 246 -538 246 -5 0 -24 6 -41
          14 -52 23 -212 67 -422 115 -215 49 -655 74 -910 51z m880 -1105 c178 -79 281
          -232 280 -420 -2 -239 -179 -426 -421 -442 -182 -12 -356 91 -434 260 -54 114
          -51 277 7 391 54 106 190 211 306 236 74 16 195 4 262 -25z m134 -1280 c17
          -20 31 -44 31 -53 0 -20 -21 -75 -94 -247 -26 -60 -65 -153 -87 -205 -22 -52
          -44 -104 -49 -115 -5 -11 -30 -69 -55 -130 -26 -60 -50 -119 -55 -130 -5 -11
          -30 -69 -55 -130 -25 -60 -61 -146 -80 -190 -19 -44 -52 -125 -75 -180 -43
          -105 -87 -212 -126 -305 -104 -252 -114 -305 -61 -344 51 -38 297 -32 445 10
          43 13 59 14 82 4 46 -19 59 -50 76 -187 9 -70 18 -143 21 -161 7 -44 -16 -76
          -85 -115 -62 -35 -160 -77 -179 -77 -7 0 -39 -8 -70 -18 -187 -60 -426 -96
          -643 -97 -171 0 -176 1 -226 27 -60 32 -107 82 -125 135 -19 52 -17 231 3 313
          9 36 25 99 36 140 51 185 65 239 77 290 7 30 22 90 34 133 12 42 30 112 42
          155 12 42 27 102 35 132 23 95 84 325 104 390 25 86 5 154 -58 193 -83 52
          -185 72 -452 88 -101 6 -193 15 -203 20 -10 6 -22 21 -27 35 -13 34 -13 320 0
          344 17 31 48 43 170 61 63 10 176 27 250 39 74 12 176 28 225 35 50 8 149 24
          220 35 72 11 186 29 255 40 69 10 184 28 255 40 72 12 175 28 229 36 55 8 102
          16 104 19 3 3 22 5 43 5 30 0 43 -7 68 -35z"
          />
        </g>
      </SVG>
      <div>Savings when you buy with a Bluefin Agent</div>
    </Wrapper>
  );
};

export default Refund;
