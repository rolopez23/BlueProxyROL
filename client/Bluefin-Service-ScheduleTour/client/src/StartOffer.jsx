import React from 'react';
import { Wrapper, Button } from './styles/StartOffer-style';

const StartOffer = () => (
  <Wrapper>
    <Button>Start Offer</Button>
  </Wrapper>
);

export default StartOffer;
