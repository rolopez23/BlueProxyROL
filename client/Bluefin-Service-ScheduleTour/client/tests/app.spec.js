/* eslint-disable*/
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import axios from 'axios';

import App from '../src/app';
import CalendarContainer from '../src/CalendarContainer';
import CalendarEntry from '../src/CalendarEntry';
import ScheduleButton from '../src/ScheduleButton';
import Contact from '../src/Contact';
import { Box1, Box2 } from '../src/styles/Contact-style';
import Refund from '../src/Refund';
import StartOffer from '../src/StartOffer';
import RefundFlyout from '../src/RefundFlyout';

jest.mock('axios');

describe('App', () => {
  let data;
  beforeEach(() => {
    data = {
      data: [
        {
          id: 123456789,
          listing_price: 123456,
          name: 'Mamba'
        }
      ]
    };
    axios.get = jest.fn(() => Promise.resolve(data));
    axios.post = jest.fn(() => Promise.resolve());
  });

  xit('snapshot renders', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    expect(app).toMatchSnapshot();
  });

  describe('On mount', () => {
    it('keeps the listingId state null if no number passed in as props', async () => {
      const wrapper = shallow(<App />);
      await Promise.all([
        expect(axios.get).toHaveBeenCalledTimes(0),
        expect(wrapper.state().houseData).toBeNull()
      ]);
    })

    it('makes a call to the api', async () => {
      const wrapper = shallow(<App listingId={123456789}/>);
      await Promise.all([
        expect(axios.get).toHaveBeenCalledWith('/house?listingId=123456789'),
        expect(axios.get).toHaveBeenCalledTimes(1)
      ]);
    });

    it('changes the listingId state to be the number passed in as props', async () => {
      const wrapper = shallow(<App listingId={123456789}/>);
      const instance = wrapper.instance();

      expect(instance.props.listingId).toEqual(123456789);
      await Promise.all([
        expect(axios.get).toHaveBeenCalledTimes(1),
        wrapper.setState({houseData: data.data[0]}, () => {
          expect(wrapper.state().houseData.id).toEqual(123456789);
          expect(wrapper.state().houseData.listing_price).toEqual(123456);
          expect(wrapper.state().houseData.name).toEqual('Mamba');
        })
      ])
    });

    // to see if rerendering works... will change as components are added
    // it('renders the name of the agent to dom', async () => {
    //     const wrapper = shallow(<App listingId={123456789}/>);
    //     await Promise.all([
    //       wrapper.setState({houseData: {
    //         id: 123456789,
    //         listing_price: 123456,
    //         name: 'Mamba'
    //       }}, () => {
    //         expect(wrapper.find('p').text()).toEqual('Mamba');
    //       })
    //     ]);
    // })
  });

  describe('Calendar', () => {
    it('should render to the dom', () => {
      const wrapper = shallow(<App listingId={123456789}/>);
      wrapper.setState({houseData: 'something'});
      expect(wrapper.find(CalendarContainer).length).toEqual(1);
    });

    it('renders calendar entries for next two weeks', () => {
      const dates = {
        day: [
          'Thursday',  'Friday',
          'Saturday',  'Sunday',
          'Monday',    'Tuesday',
          'Wednesday', 'Thursday',
          'Friday',    'Saturday',
          'Sunday',    'Monday'
        ],
        month: [
          'January',  'January',
          'January',  'February',
          'February', 'February',
          'February', 'February',
          'February', 'February',
          'February', 'February'
        ],
        date: [
          29, 30, 31, 1, 2,
           3,  4,  5, 6, 7,
           8,  9
        ]
      }
      const wrapper = mount(<CalendarContainer dates={dates}/>);
      expect(wrapper.find(CalendarEntry).length).toEqual(12);
    });

    describe('Calendar entry', () => {
      let date;
      beforeEach(() => {
        date = {
          day: 'WEDNESDAY',
          month: 'JAN',
          date: 29
        };
      });

      it('receives props of one day', () => {
        const wrapper = shallow(<CalendarEntry date={date}/>).dive();
        expect(wrapper.find('.date').text()).toEqual('29');
        expect(wrapper.find('.month').text()).toEqual('JAN');
        expect(wrapper.find('.day').text()).toEqual('WEDNESDAY');
      });

      it('Renders Card Styling for selected entry', () => {
        // eslint-disable-next-line react/jsx-filename-extension
        const wrapper = mount(<CalendarEntry date={date} selected={true} />);
        expect(wrapper).toContainMatchingElement('Selected');
        expect(wrapper).not.toContainMatchingElement('Not Selected');
        expect(wrapper).toContainMatchingElement('Day');
        expect(wrapper).toContainMatchingElement('Date');
        expect(wrapper).toContainMatchingElement('Month');
       });

       it('Renders Card Styling for selected entry', () => {
        // eslint-disable-next-line react/jsx-filename-extension
        const wrapper = mount(<CalendarEntry date={date} selected={false} />);
        expect(wrapper).toContainMatchingElement('NotSelected');
        expect(wrapper).not.toContainMatchingElement('Selected');
        expect(wrapper).toContainMatchingElement('Day');
        expect(wrapper).toContainMatchingElement('Date');
        expect(wrapper).toContainMatchingElement('Month');
       });
    });
  });


  describe('Schedule Tour Button', () => {
    it('sends a post request when clicked', async () => {
      const wrapper = mount(<App listingId={123456789}/>);
      wrapper.find(ScheduleButton).simulate('click');

      await expect(axios.post).toHaveBeenCalledTimes(1);
    });

    it('Renders Button Styling', () => {
      // eslint-disable-next-line react/jsx-filename-extension
      const wrapper = mount(<ScheduleButton />);
      expect(wrapper).toContainMatchingElement('Button');
    });
  });

  describe('Ask Button', () => {
    it('ask button renders', () => {
      const wrapper = shallow(<Contact />);
      expect(wrapper.find('AskQuestion').length).toEqual(1);
    });

    it('Renders Button Styling', () => {
      // eslint-disable-next-line react/jsx-filename-extension
      const wrapper = mount(<Contact />);
      expect(wrapper).toContainMatchingElement('Wrapper');
      expect(wrapper).toContainMatchingElement('AskQuestion');
     });
  });

  describe('Contact Button', () => {
    it('contact button renders', () => {
      const wrapper = shallow(<Contact />);
      expect(wrapper.find('PhoneNumber').length).toEqual(1);
    });

    it('Renders Button Styling', () => {
      // eslint-disable-next-line react/jsx-filename-extension
      const wrapper = mount(<Contact />);
      expect(wrapper).toContainMatchingElement('Wrapper');
      expect(wrapper).toContainMatchingElement('PhoneNumber');
     });
  });

  describe('Refund Info', () => {
    test('refund component renders', () => {
      const wrapper = shallow(<Refund />);
      expect(wrapper).toContainMatchingElement('Refund');
      expect(wrapper).toContainMatchingElement('Text');
      expect(wrapper).toContainMatchingElement('SVG');
    });

    it('calculates a fixed percent refund', () => {
      const wrapper = mount(<Refund price={10000} />);
      expect(wrapper.find('Text').text()).toEqual('Bluefin Refund: $37')
    });

    describe('Flyout', () => {
      it('renders upon click to the SVG', () => {
        const wrapper = mount(<Refund />);
        expect(wrapper.find(RefundFlyout).length).toEqual(0);
        wrapper.find('SVG').simulate('click');
        expect(wrapper.find(RefundFlyout).length).toEqual(1);
      });

      it('unrenders upon click to the SVG', () => {
        const wrapper = mount(<Refund />);
        expect(wrapper.find(RefundFlyout).length).toEqual(0);
        wrapper.find('SVG').simulate('click');
        expect(wrapper.find(RefundFlyout).length).toEqual(1);
        wrapper.find('SVG').simulate('click');
        expect(wrapper.find(RefundFlyout).length).toEqual(0);
      });
    });
  });

  describe('Start Offer button', () => {
    test('start offer component renders', () => {
      const wrapper = shallow(<StartOffer />);
      expect(wrapper).toContainMatchingElement('StartOfferButton');
      expect(wrapper).toContainMatchingElement('StartOfferText');
    });
  });
});
