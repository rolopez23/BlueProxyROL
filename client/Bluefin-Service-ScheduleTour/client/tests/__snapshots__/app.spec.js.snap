// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="sc-AykKC dvCfMh"
>
  <h3
    className="sc-AykKD geicwn"
  >
    Go Tour This Home
  </h3>
  <div
    className="sc-AykKF dXmwjQ"
  >
    <p>
      Hello
    </p>
  </div>
  <div
    className="sc-fzXfLP itsnao"
    onClick={[Function]}
  >
    Schedule Tour
  </div>
  <div
    className="sc-AykKE dVMeRz"
  >
    It's free, with no obligation - cancel anytime
  </div>
</div>
`;
